packages/host/app/components/operator-mode/interact-submode.gts

```diff
@@ -380,32 +382,34 @@ export default class InteractSubmode extends Component<Signature> {
   }
 
   private findCardInStack(card: CardDef, stackIndex: number): StackItem {
-    let item = this.stacks[stackIndex].find((item: StackItem) =>
-      card.id ? item.card.id === card.id : isEqual(item.card, card),
+    let item = this.stacks[stackIndex].find(
+      (item: StackItem) => item.url === card.id,
     );
     if (!item) {
       throw new Error(`Could not find card ${card.id} in stack ${stackIndex}`);
     }
     return item;
   }
 
-  private close = task(async (item: StackItem) => {
+  private close = (item: StackItem) => {
     // close the item first so user doesn't have to wait for the save to complete
     this.operatorModeStateService.trimItemsFromStack(item);
-    if (item.cardError) {
-      return;
-    }
+    // if (item.cardError) {
+    //   return;
+    // }
 
-    let { card, request } = item;
+    let { request, url } = item;
 
     // only save when closing a stack item in edit mode. there should be no unsaved
     // changes in isolated mode because they were saved when user toggled between
     // edit and isolated formats
-    if (item.format === 'edit') {
-      let updatedCard = this.args.saveCard(card);
-      request?.fulfill(updatedCard);
+    if (url && item.format === 'edit') {
+      this.store.save(url); // intentionally ignoring promise so caller is not waiting for save to complete
+      request?.fulfill();
```

Comment:
I'm anxious about throwing away this promise. my instinct is telling me this will come back to bite me...